### DISCUSSION TITLE
write queue is full

### DISCUSSION THREAD

**Discussion by neilyoung** [bluenviron/mediamtx#3342] [Questions]
### Question

I'm after a problem I obvioulsy have with pretty disturbed upload links to mediamtx. UDP feed is a pain to watch (when returning from mediamtx). I also see a lot of RTP packet loss warnings in the mediamtx log for the uplink.

So I switched to an RTSP uplink, which uses TCP. The packet loss rate is now not that high, but I see issues like this in the log:

```
May 09 13:07:37 ai mediamtx[94729]: 2024/05/09 13:07:37 WAR [RTSP] [session 5d22a2d1] write queue is full
May 09 13:07:38 ai mediamtx[94729]: 2024/05/09 13:07:38 WAR [RTSP] [session 5d22a2d1] write queue is full
May 09 13:07:39 ai mediamtx[94729]: 2024/05/09 13:07:39 WAR [RTSP] [session 5d22a2d1] write queue is full
May 09 13:07:40 ai mediamtx[94729]: 2024/05/09 13:07:40 WAR [RTSP] [session 5d22a2d1] write queue is full
```

Question is: What can I do with this?

**Comment by aler9** [comment]
hello, you can increase `writeQueueSize` as written in the README:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#corrupted-frames

alternatively you have to decrease the bitrate. Or switch protocol.

**Comment by neilyoung** [comment]
Yeah, it's a while ago already, I think I did something like this. The error is obvious. Thanks

**Comment by Arcitec** [comment]
## Read this for more information about safe Write Queue Size values:

https://github.com/bluenviron/mediamtx/discussions/2940#discussioncomment-11116806

**Comment by neilyoung** [comment]
Thanks
